src/main/java/com/internet/shop/dao/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,228 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.exceptions.DataBaseProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.Role;
+import com.internet.shop.models.User;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        User user = new User();
+        Set<Role> roles = new HashSet<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM users"
+                    + " JOIN user_roles ON user_roles.user_id = users.user_id"
+                    + " JOIN roles ON  roles.role_id = user_roles.role_id"
+                    + " WHERE user_login = ? AND deleted = false";
```

Comment:
I would separate this on 2 or even 3 methods
Then the code would be more readable